### PR TITLE
Fix clang-tidy ci test

### PR DESCRIPTION
**Pull request by oschaaf**
Fixes a bunch of issues reported by clang-tidy.
Mostly mechanical, except for addition of an interface for the `Uri` class.

Previously, `/source/common/utility.h` could be found by clang-tidy while in here,
with `nighthawk` as a root repository, it can't. That couldn't be fixed without adding a cyclic
dependency. (It's nice to have regardless).

With this included, we're close to having a clean clang-tidy report.

Signed-off-by: Otto van der Schaaf <oschaaf@we-amp.com>

### CODE REVIEW

File: include/nighthawk/common/uri.h [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+#include "nighthawk/common/exception.h"
+
+namespace Nighthawk {
+
+class UriException : public NighthawkException {
+public:
+  UriException(const std::string& message) : NighthawkException(message) {}
+};
+
+class Uri {
```
Comment (htuch) [inline include/nighthawk/common/uri.h]: Since this an interface, can you add Doxgen comments on methods?

File: test/client/BUILD [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+)
+
+package(default_visibility = ["//visibility:public"])
+
+licenses(["notice"])  # Apache 2
+
+envoy_cc_test_library(
+    name = "utility_lib",
+    srcs = ["utility.cc"],
+    hdrs = [
```
Comment (htuch) [inline test/client/BUILD]: Nit: prefer `hdrs = ["utility.h"],` for single line BUILD lists.

**Comment by oschaaf** [comment]
@htuch comments addressed. I see `CI` is triggered, let's see if it's happy now.

File: include/nighthawk/common/uri.h [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+   * will be explicitly set in case it is the default for the protocol. Do not use the returned
+   * value after the Uri instance is destructed.
+   */
+  virtual absl::string_view host_and_port() const PURE;
+
+  /**
+   * @return absl::string_view containing the "host" fragment parsed uri. Do not use the
+   * returned value after the Uri instance is destructed.
+   */
+  virtual absl::string_view host_without_port() const PURE;
```
Comment (htuch) [inline include/nighthawk/common/uri.h]: Tiny nit: I'd prefer `host()` and `hostWithport()` here. But, if CI passes, I'm happy to merge and have this combined in some later PR, it's not that important.

Comment (oschaaf) [reply include/nighthawk/common/uri.h]: done

**Comment by oschaaf** [comment]
That's odd, is clang-tidy flaking? I'm not sure how the last change could have made it fail, the logs look identical to the one from the previous commit, which succeeded. 
/retest

**Comment by oschaaf** [comment]
Cool, `clang-tidy` magically passed. I wonder what made it fail in the previous attempt..

**Pull request merged**